src/firmware/app/control/python_controller/difference_equation.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+"""
+This file contains the implementation of the DifferenceEquation class which generates a difference equation based on an input discrete transfer function.
+
+"""
+
+import numpy as np
+import control as ct
+from control.matlab import *
+import matplotlib.pyplot as plt
+import collections
+import _collections_abc
+import os
+
+
+class DifferenceEquation:
```

Comment:
After offline discussion with Evan; this class encompasses both the difference equation that will be directly translated to C, and also some simulation stuff for testing that will essentially be duplicated in C. We should split this class into `DifferenceEquation`, which will just contain the actual difference equation, and `DifferenceEquationSimulator` (name tbd), which will hold a `DifferenceEquation` and manage all the history, passing previous values into the difference equation, etc.. This way there's a clear encapsulation for what will be directly translated into generated C code.

Reply:
Broke up into DifferenceEquation and DifferenceEquationSim